tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -3736,18 +3920,32 @@ private[optimizer] object OptimizerCore {
       }
     }
 
-    def unapply(preTrans: PreTransBlock): Some[(List[BindingOrStat], PreTransLocalDef)] =
+    def unapply(preTrans: PreTransBlock): Some[(List[BindingOrStat], PreTransResult)] =
       Some(preTrans.bindingsAndStats, preTrans.result)
   }
 
-  private sealed abstract class PreTransNoBlock extends PreTransform
+  private sealed abstract class PreTransResult extends PreTransform
```

Comment:
OK I'll add some doc. `PreTransGenTree` is indeed a result of a `resolveLocalDef`. `PreTransResult` is anything that can the result expression of a `PreTransBlock`. That's basically everything but `PreTransGenTree` (otherwise the `PreTransBlock` itself would become a `PreTransGenTree` and `PreTransBlock` (otherwise the two `PreTransBlock` would be flattened). So far, that leaves `PreTransLocalDef`, `-UnaryOp` and `-BinaryOp`. In the future, pretty much everything else (like `PreTransIf`) would be a `PreTransResult`.